File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;
        vector<vector<string>> ans;
        
        for(int i = 0; i < strs.size(); i++) {
            string word = strs[i];
            sort(word.begin(), word.end());
            mp[word].push_back(strs[i]);
        }
        
        for(auto it = mp.begin(); it != mp.end(); it++) {
            ans.push_back(it -> second);
        }
        
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
  const strsMap = new Map();
  const ans = [];

  for (let i = 0; i < strs.length; i++) {
    const sortedKey = strs[i].split('').sort().join('');
    const group = strsMap.get(sortedKey) ?? [];
    group.push(strs[i]);
    strsMap.set(sortedKey, group);

  }

  for (const values of strsMap.values()) {
    ans.push(values);
  }

  return ans;
};